### PR TITLE
Fixed joining workspace and made home page sync with server

### PR DESCRIPTION
**Pull request by mayor04**

### CODE REVIEW

**Review by Blazebrain — APPROVED** [review]
Looks good to me

**Review by mariamadebolahamzat — APPROVED** [review]
This is goooddd

File: lib/package/base/server-request/channels/channels_api_service.dart [hunk 1]
```diff
@@ -8,8 +9,31 @@ import '../api/http_api.dart';
...
+
+    try {
+      final res = await _api.get(
+        'v1/$orgId/channels/',
+        headers: {'Authorization': 'Bearer $token'},
+      );
+      joinedChannels = res?.data;
+      print(joinedChannels);
+    } on Exception catch (e) {
+      print(e.toString());
```
Comment (isaacadariku) [inline lib/package/base/server-request/channels/channels_api_service.dart:31]: Next PR remove the print statement and make use of logger

File: lib/package/base/server-request/channels/channels_api_service.dart [hunk 2]
```diff
@@ -8,8 +9,31 @@ import '../api/http_api.dart';
...
+
+    List joinedChannels = [];
+
+    try {
+      final res = await _api.get(
+        'v1/$orgId/channels/',
+        headers: {'Authorization': 'Bearer $token'},
+      );
+      joinedChannels = res?.data;
+      print(joinedChannels);
```
Comment (isaacadariku) [inline lib/package/base/server-request/channels/channels_api_service.dart:29]: Here too

File: lib/package/base/server-request/channels/channels_api_service.dart (hunk 1 above)
Comment (isaacadariku) [inline lib/package/base/server-request/channels/channels_api_service.dart:31]: Here too, use the `log.e()` for errors, `log.i()` for information

File: lib/package/base/server-request/dms/dms_api_service.dart [hunk 3]
```diff
@@ -7,7 +8,50 @@ import '../api/http_api.dart';
...
+    try {
+      final res = await _api.get(
+        /*'v1/sidebar?org=$orgId&user=$userId'*/
+        'https://dm.zuri.chat/api/v1/sidebar?org=614679ee1a5607b13c00bcb7&user=6146ce37845b436ea04d102d',
+        headers: {'Authorization': 'Bearer $token'},
+      );
+      joinedRooms = res?.data?['joined_rooms'];
+      print(joinedRooms);
+    } on Exception catch (e) {
+      print(e.toString());
```
Comment (isaacadariku) [inline lib/package/base/server-request/dms/dms_api_service.dart:32]: Here too

File: lib/package/base/server-request/dms/dms_api_service.dart [hunk 4]
```diff
@@ -7,7 +8,50 @@ import '../api/http_api.dart';
...
+    List joinedRooms = [];
+
+    try {
+      final res = await _api.get(
+        /*'v1/sidebar?org=$orgId&user=$userId'*/
+        'https://dm.zuri.chat/api/v1/sidebar?org=614679ee1a5607b13c00bcb7&user=6146ce37845b436ea04d102d',
+        headers: {'Authorization': 'Bearer $token'},
+      );
+      joinedRooms = res?.data?['joined_rooms'];
+      print(joinedRooms);
```
Comment (isaacadariku) [inline lib/package/base/server-request/dms/dms_api_service.dart:30]: Here too

File: lib/package/base/server-request/dms/dms_api_service.dart [hunk 5]
```diff
@@ -7,7 +8,50 @@ import '../api/http_api.dart';
...
+
+  //Todo: this method does not belong here
+  Future<String?> getUser(ids) async {
+    String id = ids[0];
+    try {
+      if (ids[0] == _userService.userId) {
+        id = id[1];
+      }
+      final res = _userApi.get('users/$id');
+      print(res);
```
Comment (isaacadariku) [inline lib/package/base/server-request/dms/dms_api_service.dart:47]: here too

File: lib/package/base/server-request/dms/dms_api_service.dart [hunk 6]
```diff
@@ -7,7 +8,50 @@ import '../api/http_api.dart';
...
+    } on Exception catch (e) {
+      print(e.toString());
```
Comment (isaacadariku) [inline lib/package/base/server-request/dms/dms_api_service.dart:49]: Here too

File: lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart [hunk 7]
```diff
@@ -38,22 +38,23 @@ class WorkspaceViewModel extends BaseViewModel {
...
+    try {
       setBusy(true);
-      workspaces = await api.fetchListOfOrganizations();
-      filterWorkspace();
+      workspaces = await api.getJoinedOrganizations();
+      // filterWorkspace();
+
       setBusy(false);
     } catch (e) {
       print(e.toString());
```
Comment (isaacadariku) [inline lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart:60]: Here too

**Review by isaacadariku — COMMENTED** [review]
For all print statement, change them and use the stacked logger with `log.e()` for errors,` log.i()` for information.

**Review by isaacadariku — COMMENTED** [review]
For all print statement, change them and use the stacked logger with `log.e()` for errors,` log.i()` for information.

**Pull request merged**